cmake/utils.cmake

```diff
@@ -121,3 +121,39 @@ function(get_target_property_list target outvar)
     endforeach()
     set(${outvar} ${${outvar}} PARENT_SCOPE)
 endfunction()
+
+# Some of our targets need to be linked for real. e.g. protoc, llvm-tlbgen and all its dependencies.
+macro(set_original_launchers_if_needed)
+    if(ENABLE_DUMMY_LAUNCHERS)
+        set(CMAKE_CXX_COMPILER_LAUNCHER ${ORIGINAL_CMAKE_CXX_COMPILER_LAUNCHER})
+        set(CMAKE_C_COMPILER_LAUNCHER ${ORIGINAL_CMAKE_C_COMPILER_LAUNCHER})
+        set(CMAKE_CXX_LINKER_LAUNCHER ${ORIGINAL_CMAKE_CXX_LINKER_LAUNCHER})
+        set(CMAKE_C_LINKER_LAUNCHER ${ORIGINAL_CMAKE_C_LINKER_LAUNCHER})
+        set(LINKER_NAME ${ORIGINAL_LINKER_NAME})
+        set(USING_DUMMY_LAUNCHERS 0)
+
+        # Include again the tools.cmake file to set the real launchers for all tools
+        include(${CMAKE_SOURCE_DIR}/cmake/tools.cmake)
+    endif()
+endmacro()
+
+# Set dummy linkers to avoid unnecessary work.
+macro(set_dummy_launchers_if_needed)
+    if(ENABLE_DUMMY_LAUNCHERS)
+        set(ORIGINAL_CMAKE_CXX_COMPILER_LAUNCHER ${CMAKE_CXX_COMPILER_LAUNCHER})
```

Comment:
If set_dummy_launchers_if_needed is called twice we will overwrite ORIGINAL_CMAKE_C_COMPILER_LAUNCHER forever. Please, add a check to if that USING_DUMMY_LAUNCHERS is not set yet

Reply:
Yep, that's an excellent idea to prevent messing up with this in the future :+1: